qiskit/chemistry/results/electronic_structure_result.py

```diff
@@ -0,0 +1,346 @@
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The electronic structure result."""
+
+from typing import List, Optional, Tuple, cast
+
+import logging
+import numpy as np
+
+from qiskit.aqua.algorithms import AlgorithmResult
+from qiskit.chemistry import QMolecule
+
+from .eigenstate_result import EigenstateResult
+
+logger = logging.getLogger(__name__)
+
+# A dipole moment, when present as X, Y and Z components will normally have float values for all
+# the components. However when using Z2Symmetries, if the dipole component operator does not
+# commute with the symmetry then no evaluation is done and None will be used as the 'value'
+# indicating no measurement of the observable took place
+DipoleTuple = Tuple[Optional[float], Optional[float], Optional[float]]
+
+
+class ElectronicStructureResult(EigenstateResult):
+    """The electronic structure result."""
+
+    @property
+    def algorithm_result(self) -> AlgorithmResult:
+        """ Returns raw algorithm result """
+        return self.get('algorithm_result')
+
+    @algorithm_result.setter
+    def algorithm_result(self, value: AlgorithmResult) -> None:
+        """ Sets raw algorithm result """
+        self.data['algorithm_result'] = value
+
+    @property
+    def hartree_fock_energy(self) -> float:
+        """ Returns Hartree-Fock energy """
+        return self.get('hartree_fock_energy')
+
+    @hartree_fock_energy.setter
+    def hartree_fock_energy(self, value: float) -> None:
+        """ Sets Hartree-Fock energy """
+        self.data['hartree_fock_energy'] = value
+
+    @property
+    def nuclear_repulsion_energy(self) -> Optional[float]:
+        """ Returns nuclear repulsion energy when available from driver """
+        return self.get('nuclear_repulsion_energy')
+
+    @nuclear_repulsion_energy.setter
+    def nuclear_repulsion_energy(self, value: float) -> None:
+        """ Sets nuclear repulsion energy """
+        self.data['nuclear_repulsion_energy'] = value
+
+    @property
+    def nuclear_dipole_moment(self) -> Optional[DipoleTuple]:
+        """ Returns nuclear dipole moment X,Y,Z components in A.U when available from driver """
+        return self.get('nuclear_dipole_moment')
+
+    @nuclear_dipole_moment.setter
+    def nuclear_dipole_moment(self, value: DipoleTuple) -> None:
+        """ Sets nuclear dipole moment in A.U """
+        self.data['nuclear_dipole_moment'] = value
+
+    # TODO we need to be able to extract the statevector or the optimal parameters that can
+    # construct the circuit of the GS from here (if the algorithm supports this)
+
+    @property
+    def energy(self) -> Optional[float]:
+        """ Returns ground state energy if nuclear_repulsion_energy is available from driver """
+        # TODO the fact that this property is computed on the fly breaks the `.combine()`
```

Comment:
I am not sure why it breaks combine - the basic data that this accesses if just off the other getters, which would be combined. This is more derived quantities that a user could do but were done here to facilitate things for them.

Reply:
I am just mentioning what I noticed while writing the code. This is also not relevant for this PR.

Reply:
Thing is when they are TODOs its hard to tell!